data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -107,6 +107,7 @@ public void doInitialize() {
         doInitializeInternal();
     } catch (IOException e) {
         LOG.warn("Failed to initialize OpenSearch sink, retrying. Error - " + e.getCause());
+        closeFiles();
```

Comment:
How do we retry if we close the file and the client?

Reply:
@dlvenable , `doInitialize()` will reopen the file (when `openSearchSinkConfig.getRetryConfiguration().getDlqFile();` is called) and client connection (when `restHighLevelClient = openSearchSinkConfig.getConnectionConfiguration().createClient();` is called)

Reply:
Thanks for clarifying. I think we can probably change this in the future. I think failing to connect to the OpenSearch cluster need not force the client to close. But, this should work for now.